test/testsprite.c

```diff
@@ -48,20 +45,12 @@ static SDL_bool suspend_when_occluded;
 /* -1: infinite random moves (default); >=0: enables N deterministic moves */
 static int iterations = -1;
 
-static int done;
-
-/* Call this instead of exit(), so we can clean up SDL: atexit() is evil. */
-static void
-quit(int rc)
+void SDL_AppQuit(void)
```

Comment:
This is a void-function. 
Shouldn't an app be able to return a non-zero return value to signal something went wrong?

Reply:
SDL_AppQuit() runs right before the process exits, there's really no error you can report here that's useful.

Reply:
Isn't the process return code useful?

Reply:
For this, it's a simple success (zero on unix) or failure (one), based on whether you returned -1 or 1 from one of the callbacks.